pkg/master/storage_rbac.go

```diff
@@ -91,3 +98,42 @@ func (p RBACRESTStorageProvider) v1alpha1Storage(apiResourceConfigSource generic
 	}
 	return storage
 }
+
+func (p RBACRESTStorageProvider) PostStartHook(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) func() error {
+	var directClusterRoleAccess *clusterroleetcd.REST
+
+	if apiResourceConfigSource.AnyResourcesForVersionEnabled(rbacapiv1alpha1.SchemeGroupVersion) {
+		if apiResourceConfigSource.ResourceEnabled(rbacapiv1alpha1.SchemeGroupVersion.WithResource("clusterroles")) {
+			directClusterRoleAccess = clusterroleetcd.NewREST(restOptionsGetter(rbac.Resource("clusterroles")))
+		}
+	}
+
+	if directClusterRoleAccess == nil {
+		return func() error { return nil }
+	}
+
+	return func() error {
+		ctx := api.NewContext()
+
+		existingClusterRoles, err := directClusterRoleAccess.List(ctx, &api.ListOptions{})
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("unable to initialize clusterroles: %v", err))
+			return nil
+		}
+		// if clusterroles already exist, then assume we don't have work to do
```

Comment:
Honestly I'm really dubious about this whole methodology. Does this mean the server is in some invalid state until this runs the first time? How do clients know it is in a valid state.